test/test_packet.py

```diff
@@ -12,39 +12,53 @@
 from litex.soc.interconnect.stream import *
 from litex.soc.interconnect.packet import *
 
-packet_header_length = 31
-packet_header_fields = {
-    "field_8b"  : HeaderField(0,  0,   8),
-    "field_16b" : HeaderField(1,  0,  16),
-    "field_32b" : HeaderField(3,  0,  32),
-    "field_64b" : HeaderField(7,  0,  64),
-    "field_128b": HeaderField(15, 0, 128),
-}
-packet_header = Header(
-    fields           = packet_header_fields,
-    length           = packet_header_length,
-    swap_field_bytes = True)
-
-def packet_description(dw):
-    param_layout = packet_header.get_layout()
-    payload_layout = [("data", dw)]
-    return EndpointDescription(payload_layout, param_layout)
-
-def raw_description(dw):
-    payload_layout = [("data", dw)]
-    return EndpointDescription(payload_layout)
```

Comment:
All of this is moved into the test method as keeping it being a global variable caused the tests to fail depending on the order in which they were executed. Presumably some of these variables keep some test-modified state, but I didn't investigate this further.